File: web/src/utils/closeNode.ts
import type BlockParserState from "../types/BlockParserState";
import type MarkdownNode from "../types/MarkdownNode";

export default function closeNode(state: BlockParserState, node: MarkdownNode): void {
	node.length = state.i - node.index;
	let rule = state.rulesMap.get(node.type)!;
	if (rule.closeNode !== undefined) {
		rule.closeNode(state, node);
	}
}

File: zig/src/utils/closeNode.zig
const std = @import("std");

pub fn closeNode(
    state: *@import("../types/BlockParserState.zig").BlockParserState,
    node: *@import("../types/MarkdownNode.zig").MarkdownNode,
) void {
    node.length = state.i - node.index;

    const rule = state.rulesMap.get(node.type);
    if (rule) |r| {
        if (r.closeNode) |closeFn| {
            closeFn(state, node);
        }
    }
}
